conversion/s2s_components/codec.py

```
@@ -0,0 +1,457 @@
+
+
+def remap_key(pt_key: str) -> str:
+    if pt_key.startswith("encoder.layers."):
+        rest = pt_key[len("encoder.layers.") :]
+        idx_s, _, sub = rest.partition(".")
+        idx = int(idx_s)
+        spec = ENC_LAYOUT[idx]
+        if spec[0] == "blk":
+            return _remap_block("enc", spec[1], sub)
+        # Non-block layers carry a single ``weight`` (sometimes ``bias`` too).
+        return f"enc.{spec[0]}.{sub}" if spec[0] == "ds" else f"enc.{spec[0]}.{sub}"
```

Ruff (0.16.3)

[warning] Useless `if`-`else` condition

(RUF034)